### PR TITLE
refactor!: shares DTOs between the front and the back

### PR DESCRIPTION
**Pull request by fyusuf-a**
The DTOs, previously in the backend folder, are moved to './dtos' and adjustements were made so that they are used by both the backend and the frontend.

Some major refactoring has had to take place, notably for the dto files to comply with linting rules of the frontend.

### CODE REVIEW

File: frontend/src/components/Profile/MyFriends.vue [hunk 1]
```diff
@@ -25,14 +25,8 @@
 <script lang="ts">
 import axios from "axios";
```
Comment (tmorris42) [inline frontend/src/components/Profile/MyFriends.vue:26]:
```suggestion
import axios from 'axios';
```

File: frontend/src/components/Profile/MyFriends.vue [hunk 2]
```diff
@@ -25,14 +25,8 @@
...
 import Vue from "vue";
```
Comment (tmorris42) [inline frontend/src/components/Profile/MyFriends.vue:27]:
```suggestion
import Vue from 'vue';
```

File: frontend/src/components/Profile/MyFriends.vue [hunk 3]
```diff
@@ -25,14 +25,8 @@
...
-
-interface ResponseFriendshipDto {
-  targetId: number;
-}
-
-interface PageDto<T> {
-  data: T[];
-}
+import { ResponseFriendshipDto } from "@dtos/friendships";
```
Comment (tmorris42) [inline frontend/src/components/Profile/MyFriends.vue:28]:
```suggestion
import { ResponseFriendshipDto } from '@dtos/friendships';
```

File: frontend/src/components/Profile/MyFriends.vue [hunk 4]
```diff
@@ -25,14 +25,8 @@
...
+import { PageDto } from "@dtos/pages";
```
Comment (tmorris42) [inline frontend/src/components/Profile/MyFriends.vue:29]:
```suggestion
import { PageDto } from '@dtos/pages';
```

**Review by tmorris42 — CHANGES_REQUESTED** [review]
I see there are some check warnings due to the change to single quotation marks. I marked the ones in the changes, but some were in unchanged files, and I couldn't make suggestions on them

Otherwise, this looks really good!

File: frontend/src/components/Profile/MyFriends.vue (hunk 4 above)
Comment (fyusuf-a) [reply frontend/src/components/Profile/MyFriends.vue:29]: Thank you, I had to rebase with master, they should have come from this.

**Review by tmorris42 — APPROVED** [review]

**Pull request merged**